lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,436 @@
+              context.verticalBox(Spacing.xxSmall),
+            ],
+          ).withSymmetricalPadding(context, horizontal: Spacing.medium),
+        ),
+        // Only show toolbar action items when the field has focus or when the user has an active selection.
+        if (isFocused.value || hasSelection.value) ...[
+          _RichTextFieldToolbar(controller),
+          SizedBox(height: MediaQuery.of(context).viewInsets.bottom),
+        ]
+      ],
+    );
+  }
+}
+
+/// Provides a custom styled and reusable [quill.QuillEditor].
+///
+/// Ideally this wrapper would not be necessary but `quill` forces us to provide a set of attributes when building a
+/// [quill.QuillEditor] that are shared between this file editor instances.
+///
+/// It basically reduces boilerplate.
+class _ThemedEditor extends ConsumerWidget {
+  const _ThemedEditor({
+    required this.controller,
+    required this.backgroundColor,
+    required this.placeholder,
+    this.focus,
+    this.showCursor = true,
+    this.readonly = false,
+  });
+
+  final quill.QuillController controller;
+  final Color backgroundColor;
+  final String placeholder;
+  final FocusNode? focus;
+  final bool showCursor;
+  final bool readonly;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+    final textTheme = Theme.of(context).textTheme;
+
+    // Default [quill.DefaultTextBlockStyle] vertical and line spacings.
+    const zeroTuple = Tuple2<double, double>(0, 0);
+
+    return quill.QuillEditor(
+      controller: controller,
+      scrollController: ScrollController(),
+      scrollable: true,
+      focusNode: focus ?? FocusNode(),
+      autoFocus: !readonly,
+      readOnly: readonly,
+      expands: false,
+      padding: EdgeInsets.zero,
+      enableInteractiveSelection: !readonly,
+      showCursor: showCursor,
+      // TODO(ggirotto): Placeholder is crashing. This is a problem related to `FlutterQuill`.
+      // TODO(ggirotto): https://github.com/singerdmx/flutter-quill/issues/348
+      // placeholder: placeholder,
+      customStyles: quill.DefaultStyles(
+        paragraph: quill.DefaultTextBlockStyle(textTheme.bodyText2!, zeroTuple, zeroTuple, null),
+        placeHolder: quill.DefaultTextBlockStyle(
+          textTheme.bodyText1!.copyWith(color: theme.neutralSwatch.shade400),
+          zeroTuple,
+          zeroTuple,
+          null,
+        ),
+        code: quill.DefaultTextBlockStyle(
+          textTheme.bodyText1!,
+          zeroTuple,
+          zeroTuple,
+          BoxDecoration(color: backgroundColor),
+        ),
+      ),
+    );
+  }
+}
+
+/// A toolbar of actions to customize the editor content.
+class _RichTextFieldToolbar extends HookConsumerWidget {
+  _RichTextFieldToolbar(this.controller);
+
+  final quill.QuillController controller;
+
+  /// Maps [quill.Attribute] to its respective asset.
+  final Map<quill.Attribute<dynamic>, String> _toolBarAsset = {
```

Comment:
maybe a `Map. unmodifiable`?

Reply:
Updated. Syntax is a little bit worse but it's worth it due to the immutability from the map